core/src/main/java/com/ritense/valtimo/config/CustomTenantIdProvider.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2015-2020 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.valtimo.config;
+
+import org.camunda.bpm.engine.IdentityService;
+import org.camunda.bpm.engine.impl.cfg.multitenancy.TenantIdProvider;
+import org.camunda.bpm.engine.impl.cfg.multitenancy.TenantIdProviderCaseInstanceContext;
+import org.camunda.bpm.engine.impl.cfg.multitenancy.TenantIdProviderHistoricDecisionInstanceContext;
+import org.camunda.bpm.engine.impl.cfg.multitenancy.TenantIdProviderProcessInstanceContext;
+import org.camunda.bpm.engine.impl.context.Context;
+import org.camunda.bpm.engine.impl.identity.Authentication;
+
+import java.util.List;
+
+public class CustomTenantIdProvider implements TenantIdProvider {
+
+    @Override
+    public String provideTenantIdForProcessInstance(TenantIdProviderProcessInstanceContext ctx) {
+        return getTenantIdOfCurrentAuthentication();
+    }
+
+    @Override
+    public String provideTenantIdForCaseInstance(TenantIdProviderCaseInstanceContext ctx) {
+        return getTenantIdOfCurrentAuthentication();
+    }
+
+    @Override
+    public String provideTenantIdForHistoricDecisionInstance(TenantIdProviderHistoricDecisionInstanceContext ctx) {
+        return getTenantIdOfCurrentAuthentication();
+    }
+
+    protected String getTenantIdOfCurrentAuthentication() {
+
+        IdentityService identityService = Context.getProcessEngineConfiguration().getIdentityService();
```

Comment:
This class is from the official docs of Camunda because it's instantiated via SPI now bean support is possible I concluded. 

#### Camunda background info
- https://docs.camunda.org/manual/7.18/user-guide/process-engine/multi-tenancy/
- SPI TenantProvider did the trick as manual saving the tenants on application startup.